js/ui/screenShield.js

```diff
@@ -155,27 +155,30 @@ var ScreenShield = class {
                 return;
             }
 
-            // A new valid code unlocked the machine and user has no
-            // password set -> Go straight to the user's session.
-            if (user.password_mode == AccountsService.UserPasswordMode.NONE) {
-                this.deactivate(false);
-                return;
-            }
-
-            let paygExpectedMode = this._isGreeter ? 'gdm-unlock-dialog-payg' : 'unlock-dialog-payg';
-            if (Main.sessionMode.currentMode === 'unlock-dialog-payg') {
-                // This is the most common case.
-                Main.sessionMode.popMode(paygExpectedMode);
-            }
-
-            // The machine is unlocked but we still need to unlock the
-            // user's session with the password, so don't deactivate yet.
-            if (this._dialog) {
-                this._dialog.destroy();
-                this._dialog = null;
-            }
-
-            this.showDialog();
+            // Take the dialog instance and wait for the success message to be shown
+            this._dialog.connect('success-message-shown', () => {
+                // A new valid code unlocked the machine and user has no
+                // password set -> Go straight to the user's session.
+                if (user.password_mode == AccountsService.UserPasswordMode.NONE) {
+                    this.deactivate(false);
+                    return;
+                }
+
+                let paygExpectedMode = this._isGreeter ? 'gdm-unlock-dialog-payg' : 'unlock-dialog-payg';
+                if (Main.sessionMode.currentMode === paygExpectedMode) {
+                    // This is the most common case.
+                    Main.sessionMode.popMode(paygExpectedMode);
+                }
```

Comment:
What happened to the `else if (Main.sessionMode.currentMode == 'lock-screen')` which is here on the `Release_3.7.8` tag?

Reply:
There is no `lock-screen` mode anymore, as of c20451c2e78ef7cb70e384cce8a7b55e37b85ce4